TiDB fails to build with gotip

## Bug Report

<!-- a step by step guide for reproducing the bug. -->
```
$ make GO=gotip server
CGO_ENABLED=1  gotip build -tags codes  -ldflags '-X "github.com/pingcap/tidb/pkg/parser/mysql.TiDBReleaseVersion=v8.2.0-alpha-246-gbdb0545f73" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBBuildTS=2024-05-28 07:11:04" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitHash=bdb0545f73349ac5ce0a96f9397500d3f3e9a58f" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBGitBranch=master" -X "github.com/pingcap/tidb/pkg/util/versioninfo.TiDBEdition=Community" ' -o bin/tidb-server ./cmd/tidb-server
# github.com/pingcap/tidb/cmd/tidb-server
link: github.com/pingcap/tidb/pkg/util/printer: invalid reference to runtime.buildVersion
make: *** [Makefile:201: server] Error 1
```

--- a/pkg/util/printer/printer.go
+++ b/pkg/util/printer/printer.go
@@ -18,6 +18,7 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"runtime"
 	_ "runtime" // import link package
 	_ "unsafe"  // required by go:linkname
 
@@ -28,6 +29,12 @@ import (
 	"github.com/pingcap/tidb/pkg/util/versioninfo"
 	"go.uber.org/zap"
 )
+
+var buildVersion string
+
+func init() {
+	buildVersion = runtime.Version()
+}
 
 // PrintTiDBInfo prints the TiDB version information.
 func PrintTiDBInfo() {
@@ -175,6 +182,3 @@ func GetPrintResult(cols []string, datas [][]string) (string, bool) {
 	value = append(value, getPrintDivLine(maxColLen)...)
 	return string(value), true
 }
-
-//go:linkname buildVersion runtime.buildVersion
-var buildVersion string

--- a/pkg/util/printer/printer.go
+++ b/pkg/util/printer/printer.go
@@ -19,8 +19,6 @@ import (
 	"encoding/json"
 	"fmt"
 	"runtime"
-	_ "runtime" // import link package
-	_ "unsafe"  // required by go:linkname
 
 	"github.com/pingcap/tidb/pkg/config"
 	"github.com/pingcap/tidb/pkg/parser/mysql"

